tests/unittests.c

```diff
@@ -67,11 +67,42 @@ static void cpu_freq_expect(const char *cpu_str, uint64_t expectation) {
     return;
 }
 
-static unsigned long test_kernel_task_func(void *arg) {
+static unsigned long test_kernel_task_func1(void *arg) {
     printk("CPU[%u]: Executing %s\n", smp_processor_id(), __func__);
     return _ul(arg);
 }
 
+static inline uint64_t rand47(void) {
+    return (_ul(rand()) << 16) ^ rand();
+}
+
+#define KT2_ROUNDS 100
+#define KT2_CHUNKS 10
+static unsigned long test_kernel_task_func2(void *arg) {
+    uint32_t seed;
+    asm("rdrand %0" : "=r"(seed));
```

Comment:
This should be added as a proper interface (so when we get cpuflags support we may detect availability of the instruction, etc). Please see: https://github.com/KernelTestFramework/ktf/issues/83

Maybe for now use something simpler, unless you want to implement the proper interface of course.

Reply:
What other option do I have to initiate `srand` in KTF? Afaik there is no time function or similar.

Reply:
Since actual entropy is of no importance here, you can maybe use RDTSC or some sort of (re)combination of it to generate something resembling a pseudo-random number.